sycl/test/xocc_tests/issue_related/local_memory.cpp

```diff
@@ -0,0 +1,69 @@
+// RUN: true
+
+/*
+  The aim of this test is to check that local memory compiles and works
+  reasonably! And.... it doesn't! At the moment.
+
+  Currently Local Memory appears to be allocated as Global Memory, this appear
+  to happen despite the correct address space qualification.
+
+  This requires further discussion with the HLS team, but it appears that
+  passing a local (correctly qualified with the appropriate address space)
+  pointer to the OpenCL kernel function may not be enough.
+
+   HLS in it's C++ form represents local memory as a locally scoped "stack"
+   array that will get allocated to BRAM. So perhaps this is the required route
+   to go, which would mean we need to generate a pass that will transform
+   local address space decorated pointers to a local array.. or have some
+   intermediate representation for the local data. Not a simple task!
+
+*/
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <numeric>
+#include <vector>
+
+#include "../utilities/device_selectors.hpp"
+
+using namespace cl::sycl;
+
+constexpr size_t sz = 4;
+
+class simple_local;
+
+int main() {
+  selector_defines::CompiledForDeviceSelector selector;
+  queue q {selector};
+
+  buffer<unsigned int> a(sz);
+
+  // need to actually implement a local barrier for SPIR and remodel this
+  // example to truly tell if this is appropriately using local memory.
+  // This test only really funcitonally shows that it's not using global memory
+  // in place of local memory...but it could also be implemented via private..
+  q.submit([&](handler &cgh) {
+    auto acc_g = a.get_access<access::mode::write>(cgh);
+    accessor<int, 1, access::mode::read_write, access::target::local>
+        local_mem(range<1>(sz / 2), cgh);
+
+   cgh.parallel_for<simple_local>(nd_range<1>(range<1>(sz), range<1>(sz / 2)),
+      [=](nd_item<1> item) {
+        printf("get_local_id: %d \n", item.get_local_id()[0]);
```

Comment:
OpenCL has printf! I'm not so sure about iostream! but I don't think I meant to leave this in either...

Reply:
It was a joke anyway.
By the way, I think there is nothing in the SYCL standard that says that `printf` is expected to work, even if you have an OpenCL back-end...

Reply:
I'm pretty sure you are correct there isn't anything stating it's required to work/isn't required to work, it's implementation defined I believe; I remember a discussion a few months ago on it I think. 

It isn't implemented for Intel anyway, it will flag up as a diagnostic error as printf is a variadic function. So it really shouldn't be used inside of a kernel, I left it in this case by accident as I use them for quick testing on our FPGAs as they support printf reasonably well in emulation for now!